From f46cb861034f470d926d2df39ad5aa3bcbbe102c Mon Sep 17 00:00:00 2001
From: Yazan Shhady <yazan.shhady@solid-run.com>
Date: Sun, 26 Feb 2023 14:49:20 +0200
Subject: [PATCH 3/3] modify RZ/G2LC HummingBoard dts

-rename rzg2lc-hummingboard.dts
-remove unused dts nodes from rzg2lc-hummingboard.dts

Signed-off-by: Yazan Shhady <yazan.shhady@solid-run.com>
---
 arch/arm64/boot/dts/renesas/Makefile          |   2 +-
 ...mmingbaord.dts => rzg2lc-hummingboard.dts} | 223 ++++--------------
 .../arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi | 193 +++------------
 3 files changed, 84 insertions(+), 334 deletions(-)
 rename arch/arm64/boot/dts/renesas/{rzg2lc-hummingbaord.dts => rzg2lc-hummingboard.dts} (58%)

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index 7c9884781db0..ef3e29e08034 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -88,7 +88,7 @@ dtb-$(CONFIG_ARCH_R9A07G043) += r9a07g043u11-smarc.dtb
 dtb-$(CONFIG_ARCH_R9A07G043) += r9a07g043u11-dev.dtb
 
 dtb-$(CONFIG_ARCH_R9A07G044) += r9a07g044c2-smarc.dtb
-dtb-$(CONFIG_ARCH_R9A07G044) += rzg2lc-hummingbaord.dtb
+dtb-$(CONFIG_ARCH_R9A07G044) += rzg2lc-hummingboard.dtb
 dtb-$(CONFIG_ARCH_R9A07G044) += r9a07g044c2-dev.dtb
 dtb-$(CONFIG_ARCH_R9A07G044) += r9a07g044l2-smarc.dtb
 dtb-$(CONFIG_ARCH_R9A07G044) += r9a07g044l2-dev.dtb
diff --git a/arch/arm64/boot/dts/renesas/rzg2lc-hummingbaord.dts b/arch/arm64/boot/dts/renesas/rzg2lc-hummingboard.dts
similarity index 58%
rename from arch/arm64/boot/dts/renesas/rzg2lc-hummingbaord.dts
rename to arch/arm64/boot/dts/renesas/rzg2lc-hummingboard.dts
index bf82f33ca51e..9096049b1450 100644
--- a/arch/arm64/boot/dts/renesas/rzg2lc-hummingbaord.dts
+++ b/arch/arm64/boot/dts/renesas/rzg2lc-hummingboard.dts
@@ -5,26 +5,9 @@
  * Copyright (C) 2021 Renesas Electronics Corp.
  */
 
-/dts-v1/;
-//#include "rzg2lc-sr-som.dtsi"
-
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rzg2l-pinctrl.h>
 
-/*
-/ {
-	model = "Solidrun RZ/G2LC SOM";
-	compatible = "renesas,r9a07g044c2", "renesas,r9a07g044";
-};
-*/
-
-/* ==========================================================================*/
-//#include "r9a07g044c2.dtsi" //> SoC
-/* ==========================================================================*/
-
-/* ==========================================================================*/
-//#include "rzg2lc-smarc.dtsi"
-
 /*
  * DIP-Switch SW1 setting on SoM
  * 1 : High; 0: Low
@@ -53,23 +36,23 @@ / {
 
 #define SW_I2S0_I2S1   1
 
-#include "rzg2lc-sr-som.dtsi"
-#include "rzg2lc-sr-pinfunction.dtsi"
-
 /* comment the #define statement to disable SCIF1 (SER0) on PMOD1 (CN7) */
 #define PMOD1_SER0	1
-
 /* Select function support POEGD (1: enabled, 0 : disabled) */
 #define POEGD_SUPPORT	1
 
+/dts-v1/;
+#include "rzg2lc-sr-som.dtsi"
+#include "rzg2lc-sr-pinfunction.dtsi"
+
 / {
-	aliases {
-#if (!SW_SCIF_CAN && PMOD1_SER0 && !SCIF1_GPT3)
-		serial1 = &scif1;
-#endif
-#if (!SW_I2S0_I2S1)
-		i2c2 = &i2c2;
-#endif
+  model = "Solidrun RZ/G2LC SOM";
+  compatible = "renesas,r9a07g044c2", "renesas,r9a07g044";
+
+  x1_clk: x1-clock {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <24000000>;
 	};
 
 #if (SW_I2S0_I2S1)
@@ -80,36 +63,8 @@ sound_card {
 		dais = <&i2s2_port0>;
 	};
 #endif
-	x1_clk: x1-clock {
-		compatible = "fixed-clock";
-		#clock-cells = <0>;
-		clock-frequency = <24000000>;
-	};
-};
-
-#if ((SW_SCIF_CAN || SW_RSPI_CAN) && !MTU3_PWM_SUPPORT)
-&canfd {
-	pinctrl-0 = <&can1_pins>;
-	/delete-node/ channel0;
-};
-#else
-&canfd {
-	/delete-property/ pinctrl-0;
-	/delete-property/ pinctrl-names;
-	status = "disabled";
-};
-#endif
-
-#if (!SW_I2S0_I2S1)
-&codec_dai {
-	clocks = <&versa3 3>;
-};
-
-&cpu_dai {
-	sound-dai = <&ssi0>;
 };
 
-/*
 &i2c0 {
 
   adv7535: hdmi@3d {
@@ -152,48 +107,40 @@ codec_endpoint: endpoint {
 		};
 	};
 };
-*/
 
-&i2c2 {
-	pinctrl-0 = <&i2c2_pins>;
-	pinctrl-names = "default";
-	clock-frequency = <400000>;
-
-	status = "okay";
-
-	versa3: versa3@68 {
-		compatible = "renesas,5p35023";
-		reg = <0x68>;
-		#clock-cells = <1>;
-		clocks = <&x1_clk>;
-		clock-names = "x1";
-		assigned-clocks = <&versa3 0>,
-				   <&versa3 1>,
-				   <&versa3 2>,
-				   <&versa3 3>,
-				   <&versa3 4>,
-				   <&versa3 5>;
-		assigned-clock-rates =	<12288000>, <25000000>,
-					<12000000>, <11289600>,
-					<11289600>, <24000000>;
-		clock-divider-read-only = <1>, <1>, <1>, <1>, <1>;
-		clock-flags = <2176>, <2176>, <2176>, <2052>, <2176>, <2048>;
-	};
+&adv7535 {
+	interrupt-parent = <&pinctrl>;
+	interrupts = <RZG2L_GPIO(43, 1) IRQ_TYPE_EDGE_FALLING>;
+};
 
-	wm8978: codec@1a {
-		compatible = "wlf,wm8978";
-		#sound-dai-cells = <0>;
-		reg = <0x1a>;
+&dsi0 {
+	ports {
+		port@1 {
+			dsi0_out: endpoint {
+				remote-endpoint = <&adv7535_in>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
 	};
 };
 
-&adv7535 {
-	/delete-property/ #sound-dai-cells;
+&ssi0 {
+	pinctrl-0 = <&ssi0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+
+#if (SW_I2S0_I2S1)
 	ports {
-		/delete-node/ port@2;
+		i2s2_port0: port@0 {
+			i2s2_cpu_endpoint: endpoint {
+				remote-endpoint = <&codec_endpoint>;
+				dai-format = "i2s";
+			};
+		};
 	};
-};
 #endif
+};
 
 /*
  * To enable SCIF1 (SER0) on PMOD1 (CN7), On connector board
@@ -212,23 +159,18 @@ &scif1 {
 };
 #endif
 
-&ssi0 {
-	pinctrl-0 = <&ssi0_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-
-#if (SW_I2S0_I2S1)
-	ports {
-		i2s2_port0: port@0 {
-			i2s2_cpu_endpoint: endpoint {
-				remote-endpoint = <&codec_endpoint>;
-				dai-format = "i2s";
-			};
-		};
-	};
-#endif
+#if ((SW_SCIF_CAN || SW_RSPI_CAN) && !MTU3_PWM_SUPPORT)
+&canfd {
+	pinctrl-0 = <&can1_pins>;
+	/delete-node/ channel0;
+};
+#else
+&canfd {
+	/delete-property/ pinctrl-0;
+	/delete-property/ pinctrl-names;
+	status = "disabled";
 };
+#endif
 
 #if (SW_RSPI_CAN)
 &spi1 {
@@ -241,72 +183,3 @@ &spi1 {
 &vccq_sdhi1 {
 	gpios = <&pinctrl RZG2L_GPIO(39, 1) GPIO_ACTIVE_HIGH>;
 };
-
-&adv7535 {
-	interrupt-parent = <&pinctrl>;
-	interrupts = <RZG2L_GPIO(43, 1) IRQ_TYPE_EDGE_FALLING>;
-};
-
-#if (POEGD_SUPPORT)
-&poega {
-	status = "okay";
-};
-
-&poegb {
-	status = "okay";
-};
-
-&poegc {
-	status = "okay";
-};
-
-&poegd {
-	status = "okay";
-};
-#endif
-
-#if (SCIF1_GPT3)
-&gpt3 {
-	pinctrl-0 = <&gpt3_pins>;
-	pinctrl-names = "default";
-	channel = "both_AB";
-#if (POEGD_SUPPORT)
-	poeg = <&poega &poegb &poegc &poegd>;
-#endif
-	status = "okay";
-};
-#endif
-
-#if (MTU3_PWM_SUPPORT || MTU3_PHASE_COUNTING_SUPPORT)
-&mtu3 {
-	pinctrl-0 = <&mtu3_pins>;
-	pinctrl-names = "default";
-
-#if MTU3_PWM_SUPPORT
-	/* MTU3 PWM functions setting. */
-#if MTU3_PWM_COMPLEMENTARY_SELECT
-	/* Enable PWM complementary for MTIOC3B and MTIOC3D pins. */
-	pwm_complementary = <3 1>;
-#else
-	/* Enable PWM mode 1 out for MTIOC3C pins. */
-	pwm_mode1 = <3 1>;
-#endif
-#endif
-};
-#endif
-
-#if (MTU3_PWM_SUPPORT && MTU3_PWM_COMPLEMENTARY_SELECT && POE3_SUPPORT)
-&poe3 {
-	status = "okay";
-
-	mtu3_ch34 {
-		mtu3_outputs = <0>; /* MTIOC3B-MTIOC3D */
-	};
-};
-#endif
-
-
-
-
-
-/* ==========================================================================*/
diff --git a/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi
index 448c6a8cc037..fa0b82073f91 100644
--- a/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2lc-sr-som.dtsi
@@ -6,23 +6,12 @@
  */
 
 /dts-v1/;
-// RZ/G2LC R9A07G044C2 SoC
+/* RZ/G2LC R9A07G044C2 SoC */
 #include "r9a07g044c2.dtsi"
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rzg2l-pinctrl.h>
 
-
-// Debug DTS
-#define T1 1
-#define T2 1
-//TEST-1
-#if T1
-
-/*======================================================================*/
-// #include "rzg2lc-smarc-som.dtsi" ->
-/*======================================================================*/
-
 /*
  * MTU3 macros are created to support setting pinctrl and properties in order
  * to use MTU3 functions. However, their pins on PMOD0 are mixed with SPI1,
@@ -125,6 +114,38 @@ vccq_sdhi0: regulator-vccq-sdhi0 {
 		gpios = <&pinctrl RZG2L_GPIO(39, 0) GPIO_ACTIVE_HIGH>;
 		regulator-always-on;
 	};
+
+  chosen {
+    stdout-path = "serial0:115200n8";
+  };
+
+  usb0_vbus_otg: regulator-usb0-vbus-otg {
+    compatible = "regulator-fixed";
+
+    regulator-name = "USB0_VBUS_OTG";
+    regulator-min-microvolt = <5000000>;
+    regulator-max-microvolt = <5000000>;
+  };
+
+  vccq_sdhi1: regulator-vccq-sdhi1 {
+    compatible = "regulator-gpio";
+    regulator-name = "SDHI1 VccQ";
+    regulator-min-microvolt = <1800000>;
+    regulator-max-microvolt = <3300000>;
+    gpios-states = <1>;
+    states = <3300000 1>, <1800000 0>;
+  };
+
+  hdmi-out {
+    compatible = "hdmi-connector";
+    type = "d";
+
+    port {
+      hdmi_con_out: endpoint {
+        remote-endpoint = <&adv7535_out>;
+      };
+    };
+  };
 };
 
 &eth0 {
@@ -212,6 +233,7 @@ qspi0-ctrl {
 	 * SW1[2] should be at OFF position to enable 64 GB eMMC
 	 * SW1[2] should be at position ON to enable uSD card CN3
 	 */
+
 	gpio-sd0-dev-sel-hog {
 		gpio-hog;
 		gpios = <RZG2L_GPIO(40, 2) GPIO_ACTIVE_HIGH>;
@@ -413,100 +435,11 @@ rtc: isl1208@6f {
 		compatible = "isil,isl1208";
 		reg = <0x6f>;
 
-		external-oscillator;
-	};
-};
-
-/*======================================================================*/
-// #include "rzg2lc-smarc-som.dtsi" ^^^
-/*======================================================================*/
-//TEST-1
-#endif
-
-
-
-//TEST-2
-#if T2
-
-/*======================================================================*/
-// #include "rz-smarc-common.dtsi" ->
-/*======================================================================*/
-
-/ {
-
-/*
-	aliases {
-		serial0 = &scif0;
-		i2c0 = &i2c0;
-		i2c1 = &i2c1;
-		mmc0 = &sdhi0;
-		mmc1 = &sdhi1;
-	};
-*/
-
-	chosen {
-		stdout-path = "serial0:115200n8";
-	};
-
-	snd_rzg2l: sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,bitclock-master = <&cpu_dai>;
-		simple-audio-card,frame-master = <&cpu_dai>;
-		simple-audio-card,mclk-fs = <256>;
-
-		simple-audio-card,widgets = "Microphone", "Microphone Jack";
-		simple-audio-card,routing =
-			    "L2", "Mic Bias",
-			    "R2", "Mic Bias",
-			    "Mic Bias", "Microphone Jack";
-
-		cpu_dai: simple-audio-card,cpu {
-		};
-
-		codec_dai: simple-audio-card,codec {
-			sound-dai = <&wm8978>;
-		};
-	};
-
-	usb0_vbus_otg: regulator-usb0-vbus-otg {
-		compatible = "regulator-fixed";
-
-		regulator-name = "USB0_VBUS_OTG";
-		regulator-min-microvolt = <5000000>;
-		regulator-max-microvolt = <5000000>;
-	};
-
-	vccq_sdhi1: regulator-vccq-sdhi1 {
-		compatible = "regulator-gpio";
-		regulator-name = "SDHI1 VccQ";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <3300000>;
-		gpios-states = <1>;
-		states = <3300000 1>, <1800000 0>;
-	};
-
-	hdmi-out {
-		compatible = "hdmi-connector";
-		type = "d";
-
-		port {
-			hdmi_con_out: endpoint {
-				remote-endpoint = <&adv7535_out>;
-			};
-		};
+		//external-oscillator;
 	};
 
 };
 
-&audio_clk1{
-	clock-frequency = <11289600>;
-};
-
-&audio_clk2{
-	clock-frequency = <12288000>;
-};
-
 &canfd {
 	pinctrl-0 = <&can0_pins &can1_pins>;
 	pinctrl-names = "default";
@@ -538,48 +471,7 @@ &hsusb {
 &i2c0 {
 	pinctrl-0 = <&i2c0_pins>;
 	pinctrl-names = "default";
-
 	status = "okay";
-
-  adv7535: hdmi@3d {
-		compatible = "adi,adv7535";
-		reg = <0x3d>;
-
-		avdd-supply = <&reg_1p8v>;
-		dvdd-supply = <&reg_1p8v>;
-		pvdd-supply = <&reg_1p8v>;
-		a2vdd-supply = <&reg_1p8v>;
-		v3p3-supply = <&reg_3p3v>;
-		v1p2-supply = <&reg_1p8v>;
-
-		adi,dsi-lanes = <4>;
-
-		ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			port@0 {
-				reg = <0>;
-				adv7535_in: endpoint@0 {
-					remote-endpoint = <&dsi0_out>;
-				};
-			};
-
-			port@1 {
-				reg = <1>;
-				adv7535_out: endpoint@1 {
-					remote-endpoint = <&hdmi_con_out>;
-				};
-			};
-
-			port@2 {
-				reg = <2>;
-				codec_endpoint: endpoint {
-					remote-endpoint = <&i2s2_cpu_endpoint>;
-				};
-			};
-		};
-	};
 };
 
 &i2c1 {
@@ -680,15 +572,6 @@ &du {
 
 &dsi0 {
 	status = "okay";
-
-	ports {
-		port@1 {
-			dsi0_out: endpoint {
-				remote-endpoint = <&adv7535_in>;
-				data-lanes = <1 2 3 4>;
-			};
-		};
-	};
 };
 
 &cru {
@@ -698,9 +581,3 @@ &cru {
 &mtu3 {
 	status = "okay";
 };
-
-/*======================================================================*/
-// #include "rzg2lc-smarc-common.dtsi" ^^^
-/*======================================================================*/
-//TEST-2
-#endif
-- 
2.25.1

